tests/inc/Editor/DataBinding/BlockBindingsTest.php

```diff
@@ -512,4 +499,73 @@ public function test_get_value_with_null_fallback_results_context(): void {
 		$remote_value = BlockBindings::get_value( [ 'field' => self::MOCK_OUTPUT_FIELD_NAME ], $block, 'content' );
 		$this->assertNull( $remote_value );
 	}
+
+	/**
+	 * Creates a mock query runner with the specified result.
+	 *
+	 * @param mixed $result The result to return from the query runner.
+	 */
+	private function create_mock_query_runner_with_result( mixed $result ): MockQueryRunner {
+		$mock_qr = new MockQueryRunner();
+		$mock_qr->addResult( self::MOCK_OUTPUT_FIELD_NAME, $result );
+		return $mock_qr;
+	}
+
+	/**
+	 * Creates a mock query runner with the specified results.
+	 *
+	 * @param array $results The results to return from the query runner.
+	 */
+	private function create_mock_query_runner_with_results( array $results = [] ): MockQueryRunner {
```

Comment:
This is meant to cover the case where there are no results provided at all. I didn't want to have too many hacky if statements tbh.